sdk/program/src/hash.rs

```diff
@@ -0,0 +1,177 @@
+//! The `hash` module provides functions for creating SHA-256 hashes.
+
+use crate::sanitize::Sanitize;
+use sha2::{Digest, Sha256};
+use std::{convert::TryFrom, fmt, mem, str::FromStr};
+use thiserror::Error;
+
+pub const HASH_BYTES: usize = 32;
+#[derive(
+    Serialize, Deserialize, Clone, Copy, Default, Eq, PartialEq, Ord, PartialOrd, Hash, AbiExample,
```

Comment:
Next up: `AbiExample` needs to move out of `solana-sdk` into a stand-alone crate such that it can be imported by `solana-program-sdk` (just moving `AbiExample` into `solana-program-sdk` feels wrong...)